sklearn/utils/estimator_checks.py

```diff
@@ -3228,7 +3227,13 @@ def _enforce_estimator_tags_y(estimator, y):
     # Estimators in mono_output_task_error raise ValueError if y is of 1-D
     # Convert into a 2-D y for those estimators.
     if _safe_tags(estimator, key="multioutput_only"):
-        return np.reshape(y, (-1, 1))
+        y_2d = np.repeat(y[:, np.newaxis], 2, axis=1)
+        if _safe_tags(estimator, key="multilabel"):
+            # In multilabel classification, each target should be binary.
+            # If an estimator is supporting multioutput, it will for sure
+            # support multilabel since this is a special case of multioutput.
+            return (y_2d > y_2d.mean(axis=0)).astype(np.int64)
```

Comment:
This PR adds a test that is not used yet. Does the estimators changed in https://github.com/scikit-learn/scikit-learn/pull/21839 pass with this new input?

Reply:
At least `ClassifierChain` should use this type of target and not the `mulitoutput-multiclass`.
I indeed added the tag `"multilabel"` to `ClassifierChain`.

Reply:
Since we are not running the `if _safe_tags(estimator, key="multilabel")` in this PR, I think we can remove it and then move it in https://github.com/scikit-learn/scikit-learn/pull/21839 once this PR is merged.

I think it makes sense to keep new `y_2d` to actually test multiouput targets.